nibabel/tests/test_nifti1.py

```diff
@@ -261,12 +261,20 @@ def test_freesurfer_large_vector_hack(self):
         hdr.set_data_shape((too_big-1, 1, 1))
         assert_equal(hdr.get_data_shape(), (too_big-1, 1, 1))
         # The freesurfer case
+        full_shape = (too_big, 1, 1, 1, 1, 1, 1)
+        for dim in range(3, 8):
+            expected_dim = np.array([dim, -1, 1, 1, 1, 1, 1, 1])
+            with suppress_warnings():
+                hdr.set_data_shape(full_shape[:dim])
+            assert_equal(hdr.get_data_shape(), full_shape[:dim])
+            assert_array_equal(hdr['dim'], expected_dim)
+            assert_equal(hdr['glmin'], too_big)
+        # Allow the fourth dimension to vary
         with suppress_warnings():
-            hdr.set_data_shape((too_big, 1, 1))
-        assert_equal(hdr.get_data_shape(), (too_big, 1, 1))
-        assert_array_equal(hdr['dim'][:4], [3, -1, 1, 1])
-        assert_equal(hdr['glmin'], too_big)
-        # This only works for the case of a 3D with -1, 1, 1
+            hdr.set_data_shape((too_big, 1, 1, 4))
+        assert_equal(hdr.get_data_shape(), (too_big, 1, 1, 4))
+        assert_array_equal(hdr['dim'][:5], np.array([4, -1, 1, 1, 4]))
+        # This only works when the first 3 dimensions are -1, 1, 1
         assert_raises(HeaderDataError, hdr.set_data_shape, (too_big,))
         assert_raises(HeaderDataError, hdr.set_data_shape, (too_big,1))
```

Comment:
I think PEP8 is going to complain about the lack of spaces between values in the tuple.